packages/svelte/src/compiler/phases/3-transform/client/visitors/template.js

```diff
@@ -627,6 +627,9 @@ function serialize_inline_component(node, component_name, context) {
 	/** @type {Array<import('estree').Property[] | import('estree').Expression>} */
 	const props_and_spreads = [];
 
+	/** @type {(import('estree').Expression | null)[]} */
+	const spreads_keys = [];
```

Comment:
all this stuff is unnecessary! we can create the derived inside `spread_props`

Reply:
Duh 🤦🏻‍♂️ completely missed this